File: php/features.php
<?php
declare(strict_types=1);

// Kmail SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KmailFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KmailBaseFeature();
            case "test":
                return new KmailTestFeature();
            default:
                return new KmailBaseFeature();
        }
    }
}

File: py/kmail_sdk/features.py
# Kmail SDK feature factory

from kmail_sdk.feature.base_feature import KmailBaseFeature
from kmail_sdk.feature.test_feature import KmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KmailBaseFeature(),
        "test": lambda: KmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
